lib/mayaUsd/ufe/UsdMaterial.cpp

```diff
@@ -0,0 +1,125 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdMaterial.h"
+
+#include <mayaUsd/ufe/Utils.h>
+
+#include <ufe/types.h>
+
+#include <stdexcept>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+UsdMaterial::UsdMaterial(const UsdSceneItem::Ptr& item)
+    : Ufe::Material()
+    , fItem(item)
+{
+}
+
+UsdMaterial::~UsdMaterial() { }
+
+/*static*/
+UsdMaterial::Ptr UsdMaterial::create(const UsdSceneItem::Ptr& item)
+{
+    return std::make_shared<UsdMaterial>(item);
+}
+
+std::vector<PXR_NS::UsdPrim> UsdMaterial::getConnectedShaders(const PXR_NS::UsdShadeOutput& port) const
+{
+    // Dig down across NodeGraph boundaries until a surface shader is found.
+    std::vector<PXR_NS::UsdPrim> shaders;
+    for (const auto& sourceInfo : port.GetConnectedSources()) {
+        if (sourceInfo.source.GetPrim().GetTypeName() != "Shader") {
+            if (sourceInfo.sourceType == UsdShadeAttributeType::Output) {
+                const std::vector<PXR_NS::UsdPrim> connectedShaders
+                    = getConnectedShaders(sourceInfo.source.GetOutput(sourceInfo.sourceName));
+                shaders.insert(end(shaders), begin(connectedShaders), end(connectedShaders));
+            }
+        } else {
+            shaders.push_back(sourceInfo.source.GetPrim());
+        }
+    }
+    return shaders;
+}
+
+std::vector<Ufe::SceneItem::Ptr> UsdMaterial::getMaterials() const
+{
+    PXR_NAMESPACE_USING_DIRECTIVE
+
+    // Find the material(s) attached to our SceneItem.
+
+    std::vector<Ufe::SceneItem::Ptr> materials;
+
+    if (!TF_VERIFY(fItem)) {
+        return materials;
+    }
+
+    std::vector<PXR_NS::UsdPrim> materialPrims;
+
+    const PXR_NS::UsdPrim&                            prim = fItem->prim();
+    PXR_NS::UsdShadeMaterialBindingAPI                bindingApi(prim);
+    PXR_NS::UsdShadeMaterialBindingAPI::DirectBinding directBinding = bindingApi.GetDirectBinding();
+
+    // 1. Simple case: A material is directly attached to our object.
+    const PXR_NS::UsdShadeMaterial material = directBinding.GetMaterial();
+    if (material) {
+        materialPrims.push_back(material.GetPrim());
```

Comment:
That is a UsdShadeMaterial. You must iterate GetSurfaceOuputs and getConnectedShaders to get the same type as will be found on geom subsets.